apps/kampus/app/pano/PanoFeed.tsx

```diff
@@ -0,0 +1,77 @@
+import { Suspense, useCallback } from "react";
+import { graphql, usePaginationFragment } from "react-relay";
+
+import { Button } from "@kampus/ui-next";
+
+import { PostItem } from "~/app/pano/features/post-list/PostItem";
+import { type PanoFeedFragment$key } from "./__generated__/PanoFeedFragment.graphql";
+
+const fragment = graphql`
+  fragment PanoFeedFragment on Viewer
+  @argumentDefinitions(
+    after: { type: "String" }
+    first: { type: "Int", defaultValue: 10 }
+    before: { type: "String" }
+    last: { type: "Int" }
+  )
+  @refetchable(queryName: "PanoFeedPaginationQuery") {
+    panoFeed(first: $first, after: $after, last: $last, before: $before)
+      @connection(key: "PanoFeedFragment__panoFeed") {
+      edges {
+        cursor
+        node {
+          id
+          ...PostItem_post
+        }
+      }
+    }
+  }
+`;
+
+interface Props {
+  panoFeed: PanoFeedFragment$key;
+}
+
+export function PanoFeed(props: Props) {
+  const { data, hasNext, hasPrevious, loadNext, loadPrevious } = usePaginationFragment(
```

Comment:
we write a complex query to get these helpers, i think it's a really great tradeoff.